### PR TITLE
Add a readme to approach 1.6.3 Product scenarios

### PR DESCRIPTION
**Pull request by isuruuy429**
## Purpose
The purpose of this PR is to add readme to approach 1.6.3-Replace a value of the incoming message using enrich mediator

## Approach
Following content are added to readme file

When to use
Sample use-case
Pre-requisites
Development guidelines
Deployment guidelines
Supported versions
Test cases

### CODE REVIEW

File: product-scenarios/1-integrating-systems-that-communicate-in-heterogeneous-message-formats/1.6-xml-message-enrichment/1.6.1-Modify-payload-by-adding-a-child-using-enrich-mediator/pom.xml [hunk 1]
```diff
@@ -0,0 +1,177 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (milindaperera) [inline product-scenarios/1-integrating-systems-that-communicate-in-heterogeneous-message-formats/1.6-xml-message-enrichment/1.6.1-Modify-payload-by-adding-a-child-using-enrich-mediator/pom.xml]: Let's add license headers

File: product-scenarios/1-integrating-systems-that-communicate-in-heterogeneous-message-formats/1.6-xml-message-enrichment/1.6.1-Modify-payload-by-adding-a-child-using-enrich-mediator/readme.md [hunk 2]
```diff
@@ -0,0 +1,178 @@
...
+
+## When to use
+The Enrich Mediator can process a message based on a given source configuration and then perform the specified action on the message by using the target configuration. It is often used to do slight modifications to the payload. There are main three target actions as “Replace”, “Child” and “Sibling”. In this scenario we focus on adding a child element to the payload. 
+
+
+## Sample use case
+Assume that we have our original payload to be sent as below. 
+
+original payload
+```
```
Comment (milindaperera) [inline product-scenarios/1-integrating-systems-that-communicate-in-heterogeneous-message-formats/1.6-xml-message-enrichment/1.6.1-Modify-payload-by-adding-a-child-using-enrich-mediator/readme.md]: let's use XML formatting

File: product-scenarios/1-integrating-systems-that-communicate-in-heterogeneous-message-formats/1.6-xml-message-enrichment/1.6.1-Modify-payload-by-adding-a-child-using-enrich-mediator/readme.md [hunk 3]
```diff
@@ -0,0 +1,178 @@
...
+         <enrich>
+            <source clone="true" type="inline">
+               <xsd:symbol xmlns:xsd="http://services.samples/xsd">SUN</xsd:symbol>
+            </source>
+            <target xmlns:xsd="http://services.samples/xsd"
+                    xmlns:ser="http://services.samples"
+                    action="child"
+                    xpath="//ser:getQuote/ser:request"/>
+         </enrich>
+         <log level="full"/>
```
Comment (milindaperera) [inline product-scenarios/1-integrating-systems-that-communicate-in-heterogeneous-message-formats/1.6-xml-message-enrichment/1.6.1-Modify-payload-by-adding-a-child-using-enrich-mediator/readme.md]: logging is not the best way to demonstrate samples. let's respond the enriched message

File: product-scenarios/1-integrating-systems-that-communicate-in-heterogeneous-message-formats/1.6-xml-message-enrichment/1.6.1-Modify-payload-by-adding-a-child-using-enrich-mediator/src/test/java/org/wso2/carbon/esb/endpoint/test/AuthenticateStubUtil.java [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+package org.wso2.carbon.esb.endpoint.test;
+
+import org.apache.axis2.client.Options;
+import org.apache.axis2.client.ServiceClient;
+import org.apache.axis2.client.Stub;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.wso2.carbon.utils.CarbonUtils;
+
+public class AuthenticateStubUtil {
```
Comment (milindaperera) [inline product-scenarios/1-integrating-systems-that-communicate-in-heterogeneous-message-formats/1.6-xml-message-enrichment/1.6.1-Modify-payload-by-adding-a-child-using-enrich-mediator/src/test/java/org/wso2/carbon/esb/endpoint/test/AuthenticateStubUtil.java]: we do not need this class

File: product-scenarios/1-integrating-systems-that-communicate-in-heterogeneous-message-formats/1.6-xml-message-enrichment/1.6.1-Modify-payload-by-adding-a-child-using-enrich-mediator/src/test/java/org/wso2/carbon/esb/endpoint/test/AuthenticatorClient.java [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+package org.wso2.carbon.esb.endpoint.test;
+
+import org.apache.axis2.client.Stub;
+import org.apache.axis2.context.ServiceContext;
+import org.apache.axis2.transport.http.HTTPConstants;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.wso2.carbon.authenticator.stub.AuthenticationAdminStub;
+
+public class AuthenticatorClient {
```
Comment (milindaperera) [inline product-scenarios/1-integrating-systems-that-communicate-in-heterogeneous-message-formats/1.6-xml-message-enrichment/1.6.1-Modify-payload-by-adding-a-child-using-enrich-mediator/src/test/java/org/wso2/carbon/esb/endpoint/test/AuthenticatorClient.java]: Do not need this class

File: product-scenarios/1-integrating-systems-that-communicate-in-heterogeneous-message-formats/1.6-xml-message-enrichment/1.6.1-Modify-payload-by-adding-a-child-using-enrich-mediator/src/test/java/org/wso2/carbon/esb/endpoint/test/ScenarioSampleTest.java [hunk 6]
```diff
@@ -0,0 +1,178 @@
...
+import org.wso2.carbon.integration.common.admin.client.CarbonAppUploaderClient;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.Calendar;
+import java.util.Properties;
+import javax.activation.DataHandler;
+import javax.activation.FileDataSource;
+
+public class ScenarioSampleTest extends ScenarioTestBase {
```
Comment (milindaperera) [inline product-scenarios/1-integrating-systems-that-communicate-in-heterogeneous-message-formats/1.6-xml-message-enrichment/1.6.1-Modify-payload-by-adding-a-child-using-enrich-mediator/src/test/java/org/wso2/carbon/esb/endpoint/test/ScenarioSampleTest.java]: let's extend https://github.com/wso2/product-ei/blob/product-scenarios/product-scenarios/scenarios-commons/src/main/java/org/wso2/carbon/esb/scenario/test/common/ScenarioTestBase.java

File: product-scenarios/1-integrating-systems-that-communicate-in-heterogeneous-message-formats/1.6-xml-message-enrichment/1.6.3-Replace-a-value-of-the-incoming-message-using-enrich-mediator/src/test/java/org/wso2/carbon/esb/endpoint/test/ScenarioTestBase.java [hunk 7]
```diff
@@ -0,0 +1,85 @@
...
+
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.Properties;
+
+public class ScenarioTestBase {
```
Comment (milindaperera) [inline product-scenarios/1-integrating-systems-that-communicate-in-heterogeneous-message-formats/1.6-xml-message-enrichment/1.6.3-Replace-a-value-of-the-incoming-message-using-enrich-mediator/src/test/java/org/wso2/carbon/esb/endpoint/test/ScenarioTestBase.java]: we do not need this class. we can use [1]

[1] https://github.com/wso2/product-ei/blob/product-scenarios/product-scenarios/scenarios-commons/src/main/java/org/wso2/carbon/esb/scenario/test/common/ScenarioTestBase.java

File: product-scenarios/1-integrating-systems-that-communicate-in-heterogeneous-message-formats/1.6-xml-message-enrichment/1.6.1-Modify-payload-by-adding-a-child-using-enrich-mediator/src/test/java/org/wso2/carbon/esb/endpoint/test/ScenarioTestBase.java [hunk 8]
```diff
@@ -0,0 +1,85 @@
...
+
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.Properties;
+
+public class ScenarioTestBase {
```
Comment (milindaperera) [inline product-scenarios/1-integrating-systems-that-communicate-in-heterogeneous-message-formats/1.6-xml-message-enrichment/1.6.1-Modify-payload-by-adding-a-child-using-enrich-mediator/src/test/java/org/wso2/carbon/esb/endpoint/test/ScenarioTestBase.java]: we do not need this class. this replaced with common test base class. let's remove

[1] https://github.com/wso2/product-ei/blob/product-scenarios/product-scenarios/scenarios-commons/src/main/java/org/wso2/carbon/esb/scenario/test/common/ScenarioTestBase.java

File: product-scenarios/1-integrating-systems-that-communicate-in-heterogeneous-message-formats/1.6-xml-message-enrichment/1.6.2-Modify-payload-by-adding-a-sibling-using-enrich-mediator/readme.md [hunk 9]
```diff
@@ -0,0 +1,178 @@
...
+
+## When to use
+The Enrich Mediator can process a message based on a given source configuration and then perform the specified action on the message by using the target configuration. It is often used to do slight modifications to the payload. There are main three target actions as “Replace”, “Child” and “Sibling”. In this scenario we focus on adding a sibling element to the payload. 
+
+
+## Sample use case
+Assume that we have our original payload to be sent as below. 
+
+original payload
+```
```
Comment (milindaperera) [inline product-scenarios/1-integrating-systems-that-communicate-in-heterogeneous-message-formats/1.6-xml-message-enrichment/1.6.2-Modify-payload-by-adding-a-sibling-using-enrich-mediator/readme.md]: Let's use xml formatting in all the xml documented places

File: product-scenarios/1-integrating-systems-that-communicate-in-heterogeneous-message-formats/1.6-xml-message-enrichment/1.6.2-Modify-payload-by-adding-a-sibling-using-enrich-mediator/readme.md [hunk 10]
```diff
@@ -0,0 +1,178 @@
...
+</ser:getQuote>
+</soapenv:Body>
+</soapenv:Envelope>
+```
+
+Above proxy will add the sibling element <xsd:symbol>SUN</xsd:symbol> to the <ser:request> element before the payload is sent to the endpoint. Below is the log when the proxy is called. You can see the sibling element has been added after the Enrich3 proxy is called.  
+
+
+```
+[2018-11-26 15:39:23,302] [EI-Core]  INFO - LogMediator To: /services/Enrich3, WSAction: urn:getQuote, SOAPAction: urn:getQuote, MessageID: urn:uuid:40ae063a-9eae-487c-afde-052eb2d2e940, Direction: request, Envelope: <?xml version='1.0' encoding='utf-8'?><soap:Envelope xmlns:soap="http://www.w3.org/2003/05/soap-envelope" xmlns:xsd="http://services.samples/xsd" xmlns:ser="http://services.samples"><soap:Body>
```
Comment (milindaperera) [inline product-scenarios/1-integrating-systems-that-communicate-in-heterogeneous-message-formats/1.6-xml-message-enrichment/1.6.2-Modify-payload-by-adding-a-sibling-using-enrich-mediator/readme.md]: Let's remove debug logs.
Let's try to demonstrate use cases by responding to the client

File: product-scenarios/1-integrating-systems-that-communicate-in-heterogeneous-message-formats/1.6-xml-message-enrichment/1.6.2-Modify-payload-by-adding-a-sibling-using-enrich-mediator/src/test/java/org/wso2/carbon/esb/endpoint/test/AuthenticateStubUtil.java [hunk 11]
```diff
@@ -0,0 +1,39 @@
...
+package org.wso2.carbon.esb.endpoint.test;
+
+import org.apache.axis2.client.Options;
+import org.apache.axis2.client.ServiceClient;
+import org.apache.axis2.client.Stub;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.wso2.carbon.utils.CarbonUtils;
+
+public class AuthenticateStubUtil {
```
Comment (milindaperera) [inline product-scenarios/1-integrating-systems-that-communicate-in-heterogeneous-message-formats/1.6-xml-message-enrichment/1.6.2-Modify-payload-by-adding-a-sibling-using-enrich-mediator/src/test/java/org/wso2/carbon/esb/endpoint/test/AuthenticateStubUtil.java]: remove this class

**Comment by milindaperera** [comment]
@isuruuy429 let's cleanup all the unwanted classes and reuse from [1] 

[1] https://github.com/wso2/product-ei/tree/product-scenarios/product-scenarios/scenarios-commons

File: product-scenarios/1-integrating-systems-that-communicate-in-heterogeneous-message-formats/1.6-xml-message-enrichment/1.6.1-Modify-payload-by-adding-a-child-using-enrich-mediator/pom.xml (hunk 1 above)
Comment (isuruuy429) [reply product-scenarios/1-integrating-systems-that-communicate-in-heterogeneous-message-formats/1.6-xml-message-enrichment/1.6.1-Modify-payload-by-adding-a-child-using-enrich-mediator/pom.xml]: removed all the unwanted files and added readme.md only.

File: product-scenarios/1-integrating-systems-that-communicate-in-heterogeneous-message-formats/1.6-xml-message-enrichment/1.6.1-Modify-payload-by-adding-a-child-using-enrich-mediator/readme.md (hunk 2 above)
Comment (isuruuy429) [reply product-scenarios/1-integrating-systems-that-communicate-in-heterogeneous-message-formats/1.6-xml-message-enrichment/1.6.1-Modify-payload-by-adding-a-child-using-enrich-mediator/readme.md]: Done

File: product-scenarios/1-integrating-systems-that-communicate-in-heterogeneous-message-formats/1.6-xml-message-enrichment/1.6.1-Modify-payload-by-adding-a-child-using-enrich-mediator/readme.md (hunk 3 above)
Comment (isuruuy429) [reply product-scenarios/1-integrating-systems-that-communicate-in-heterogeneous-message-formats/1.6-xml-message-enrichment/1.6.1-Modify-payload-by-adding-a-child-using-enrich-mediator/readme.md]: This is updated to use a respond mediator. Therefore removed the logging.

File: product-scenarios/1-integrating-systems-that-communicate-in-heterogeneous-message-formats/1.6-xml-message-enrichment/1.6.1-Modify-payload-by-adding-a-child-using-enrich-mediator/src/test/java/org/wso2/carbon/esb/endpoint/test/AuthenticateStubUtil.java (hunk 4 above)
Comment (isuruuy429) [reply product-scenarios/1-integrating-systems-that-communicate-in-heterogeneous-message-formats/1.6-xml-message-enrichment/1.6.1-Modify-payload-by-adding-a-child-using-enrich-mediator/src/test/java/org/wso2/carbon/esb/endpoint/test/AuthenticateStubUtil.java]: removed all the unwanted files and added readme.md only.

File: product-scenarios/1-integrating-systems-that-communicate-in-heterogeneous-message-formats/1.6-xml-message-enrichment/1.6.1-Modify-payload-by-adding-a-child-using-enrich-mediator/src/test/java/org/wso2/carbon/esb/endpoint/test/AuthenticatorClient.java (hunk 5 above)
Comment (isuruuy429) [reply product-scenarios/1-integrating-systems-that-communicate-in-heterogeneous-message-formats/1.6-xml-message-enrichment/1.6.1-Modify-payload-by-adding-a-child-using-enrich-mediator/src/test/java/org/wso2/carbon/esb/endpoint/test/AuthenticatorClient.java]: removed all the unwanted files and added readme.md only.

File: product-scenarios/1-integrating-systems-that-communicate-in-heterogeneous-message-formats/1.6-xml-message-enrichment/1.6.3-Replace-a-value-of-the-incoming-message-using-enrich-mediator/src/test/java/org/wso2/carbon/esb/endpoint/test/ScenarioTestBase.java (hunk 7 above)
Comment (isuruuy429) [reply product-scenarios/1-integrating-systems-that-communicate-in-heterogeneous-message-formats/1.6-xml-message-enrichment/1.6.3-Replace-a-value-of-the-incoming-message-using-enrich-mediator/src/test/java/org/wso2/carbon/esb/endpoint/test/ScenarioTestBase.java]: removed all the unwanted files and added readme.md only.

File: product-scenarios/1-integrating-systems-that-communicate-in-heterogeneous-message-formats/1.6-xml-message-enrichment/1.6.1-Modify-payload-by-adding-a-child-using-enrich-mediator/src/test/java/org/wso2/carbon/esb/endpoint/test/ScenarioTestBase.java (hunk 8 above)
Comment (isuruuy429) [reply product-scenarios/1-integrating-systems-that-communicate-in-heterogeneous-message-formats/1.6-xml-message-enrichment/1.6.1-Modify-payload-by-adding-a-child-using-enrich-mediator/src/test/java/org/wso2/carbon/esb/endpoint/test/ScenarioTestBase.java]: Removed all the unwanted classes.

File: product-scenarios/1-integrating-systems-that-communicate-in-heterogeneous-message-formats/1.6-xml-message-enrichment/1.6.2-Modify-payload-by-adding-a-sibling-using-enrich-mediator/readme.md (hunk 9 above)
Comment (isuruuy429) [reply product-scenarios/1-integrating-systems-that-communicate-in-heterogeneous-message-formats/1.6-xml-message-enrichment/1.6.2-Modify-payload-by-adding-a-sibling-using-enrich-mediator/readme.md]: Done

File: product-scenarios/1-integrating-systems-that-communicate-in-heterogeneous-message-formats/1.6-xml-message-enrichment/1.6.2-Modify-payload-by-adding-a-sibling-using-enrich-mediator/readme.md (hunk 10 above)
Comment (isuruuy429) [reply product-scenarios/1-integrating-systems-that-communicate-in-heterogeneous-message-formats/1.6-xml-message-enrichment/1.6.2-Modify-payload-by-adding-a-sibling-using-enrich-mediator/readme.md]: Removed all the logs.

File: product-scenarios/1-integrating-systems-that-communicate-in-heterogeneous-message-formats/1.6-xml-message-enrichment/1.6.2-Modify-payload-by-adding-a-sibling-using-enrich-mediator/src/test/java/org/wso2/carbon/esb/endpoint/test/AuthenticateStubUtil.java (hunk 11 above)
Comment (isuruuy429) [reply product-scenarios/1-integrating-systems-that-communicate-in-heterogeneous-message-formats/1.6-xml-message-enrichment/1.6.2-Modify-payload-by-adding-a-sibling-using-enrich-mediator/src/test/java/org/wso2/carbon/esb/endpoint/test/AuthenticateStubUtil.java]: Done

**Pull request merged**